src/lib.rs

```diff
@@ -2428,6 +2428,116 @@ pub fn undirected_gnm_random_graph(
     Ok(graph)
 }
 
+#[inline]
+fn pnorm(x: f64, p: f64) -> f64 {
+    if p == 1.0 || p == std::f64::INFINITY {
+        x.abs()
+    } else if p == 2.0 {
+        x * x
+    } else {
+        x.abs().powf(p)
+    }
+}
+
+fn distance(x: &[f64], y: &[f64], p: f64) -> f64 {
+    let it = x.iter().zip(y.iter()).map(|(xi, yi)| pnorm(xi - yi, p));
+
+    if p == std::f64::INFINITY {
+        it.fold(-1.0, |max, x| if x > max { x } else { max })
+    } else {
+        it.sum()
+    }
+}
+
+/// Returns a random geometric graph in the unit cube of dimensions `dim`.
+///
+/// The random geometric graph model places `num_nodes` nodes uniformly at
+/// random in the unit cube. Two nodes are joined by an edge if the
+/// distance between the nodes is at most `radius`.
+///
+/// Each node has a node attribute ``'pos'`` that stores the
+/// position of that node in Euclidean space as provided by the
+/// ``pos`` keyword argument or, if ``pos`` was not provided, as
+/// generated by this function.
+///
+/// :param int num_nodes: The number of nodes to create in the graph
+/// :param float radius: Distance threshold value
+/// :param int dim: Dimension of node positions. Default: 2
+/// :param list pos: Optional list with node positions as values
+/// :param float p: Which Minkowski distance metric to use.  `p` has to meet the condition
+///     ``1 <= p <= infinity``.
+///     If this argument is not specified, the :math:`L^2` metric
+///     (the Euclidean distance metric), p = 2 is used.
+/// :param int seed: An optional seed to use for the random number generator
+///
+/// :return: A PyGraph object
+/// :rtype: PyGraph
+#[pyfunction]
+#[text_signature = "(num_nodes, radius, dim=2, pos=None, p=2.0, seed=None, /)"]
+fn random_geometric_graph(
```

Comment:
Is there a reason for this function to be private?

Reply:
FWIW, it doesn't matter too much (at least currently, after #108 is unblocked it might be important) because at the end of the day retworkx gets compiled to a cdynlib that python loads as long as we have the `#[pyfunction]` macro we'll be building a C FFI function in that binary as long as we export it to python that via the pymodule at the bottom of lib.rs to make it importable. Having `pub` here means it's public in terms of other rust code which we don't really have any consumers for. But it also doesn't hurt either and might make it clearer to say it's public even if it's not strictly needed.